src/include/kdbmacros.h

```diff
@@ -37,24 +37,33 @@
 	do                                                                                                                                 \
 	{                                                                                                                                  \
 		if (errno == EACCES)                                                                                                       \
-			ELEKTRA_SET_ERROR (109, parentKey, strerror (errno));                                                              \
+			ELEKTRA_SET_RESOURCE_ERRORF (                                                                                      \
+				parentKey,                                                                                                 \
+				"Insufficient permissions to open configuration file for reading. Reason: %s. You might want "             \
+				"to retry as root or change access using chmod.",                                                          \
```

Comment:
Somehow mark this (maybe with TODO SOLUTION): This is a solution and should be separated.

Reply:
I would need to look through all errors again anyway so no need to add an extra TODO here